integration-tests/src/tests/completion.test.ts

```diff
@@ -0,0 +1,75 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) 2023 Savoir-faire Linux. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as assert from 'assert'
+import * as vscode from 'vscode'
+import path from 'path'
+
+async function delay (ms: number): Promise<void> {
+  await new Promise(resolve => setTimeout(resolve, ms))
+}
+
+suite('Bitbake Completion Test Suite', () => {
+  const filePath = path.resolve(__dirname, '../../project-folder/sources/meta-fixtures/completion.bb')
+  const docUri = vscode.Uri.parse(`file://${filePath}`)
+
+  suiteSetup(async function (this: Mocha.Context) {
+    this.timeout(100000)
+    const vscodeBitbake = vscode.extensions.getExtension('savoirfairelinux.bitbake')
+    if (vscodeBitbake === undefined) {
+      assert.fail('Bitbake extension is not available')
+    }
+    await vscodeBitbake.activate()
+    while (!vscodeBitbake.isActive) {
+      await delay(100)
+    }
+    const document = await vscode.workspace.openTextDocument(docUri)
+    await vscode.window.showTextDocument(document)
+  })
+
+  const checkHasItemWithLabel = (completionList: vscode.CompletionList, label: string): boolean => {
+    return completionList.items.some(item => {
+      if (typeof item.label === 'string') {
+        return item.label === label
+      }
+      return item.label.label === label
+    })
+  }
+
+  const testCompletion = async (position: vscode.Position, expected: string): Promise<void> => {
+    let completionList: vscode.CompletionList = { items: [] }
+    while (!checkHasItemWithLabel(completionList, expected)) {
+      completionList = await vscode.commands.executeCommand<vscode.CompletionList>(
```

Comment:
The command will return a CompletionList on each call. However, it takes some time for tree-sitter to analyze the documents, and for our extension to build the "embedded language documents". Until it is done, the command will return several CompletionList that don't take the "embedded language documents" into account, since they do not exist yet. I also tried to look into the folder where the "embedded language documents" are generated, and wait for them to appear before calling `vscode.executeCompletionItemProvider`. I managed to wait for them to appear before making the call, but it still was not enough, for unknown reasons. Since it was not working, I did not push that code.

Reply:
Ack